lib/OpenQA/Task/Needle/Save.pm

```diff
@@ -52,8 +53,12 @@ sub _format_git_error {
     return "<strong>Failed to save $name.</strong><br><pre>$error</pre>";
 }
 
-sub _save_needle {
-    my ($app, $minion_job, $args) = @_;
+sub _create_signal_guard ($job) {
+    return OpenQA::Task::SignalGuard->new($job);
+}
```

Comment:
```suggestion
sub _create_signal_guard ($job) { OpenQA::Task::SignalGuard->new($job) }
```

but I actually I suggest to not define this function as just using the object creation in place should be clear enough. You can still mock the module OpenQA::Task::SignalGuard for testing purposes

Reply:
Without the object creation how can I check if abort is reset to 1? I don't have the object used in the save needle!

Reply:
I think I found how to do it!